src/devices/CharacterLcd/LcdInterface.Spi.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Iot.Device.Multiplexing;
+
+namespace Iot.Device.CharacterLcd
+{
+    public abstract partial class LcdInterface : IDisposable
```

Comment:
I find this a bit confusing - if you're using SPI interface why is this not using SpiDevice?

Reply:
would it be possible to implement SpiDevice using Sn74hc595? If not consider changing name to `LcdInterface.ShiftRegister` or even explicitly `Sn74hc595`

Reply:
Generic `ShiftRegister` binding supports control using `SpiDevice` and even has an example diagram that uses the Sn74hc595: https://github.com/dotnet/iot/tree/main/src/devices/ShiftRegister#using-spi
But for some reason the `Sn74hc595` binding is missing the `SpiDevice` constructor even though it states that is supports SPI control: https://github.com/dotnet/iot/blob/main/src/devices/Sn74hc595/Sn74hc595.cs#L12. I can try to add that, maybe in a separate PR.

Then the user will be able to 'indirectly' control the Lcd using SpiDevice. e.g.:
```
using SpiDevice spiDevice = SpiDevice.Create(new(0, 0));
using Sn74hc595 sr = new(spiDevice);
using LcdInterface lcdInterface = LcdInterface.CreateSpi(registerSelectPin, enablePin, dataPins, backlightPin, sr);
using Lcd1602 lcd = new(lcdInterface);
lcd.Clear();
lcd.Write("Hello World");
```
Still changing the name to LcdInterface.Sn74hc595 might be more appropriate. Do you think following names would be better?
File name -> `LcdInterface.Sn74hc595.cs`
Inner class name -> `private class Sn74hc595LcdInterface : LcdInterface { ... }`
create instance method -> `LcdInterface.CreateSn74hc595(...)`

Reply:
Would this mechanism work with all shift registers? I assume yes, if so then I see two paths forward:
- change this to LcdInterface.ShiftRegister and use base class rather than specific ShiftRegister implementation
- create SpiDevice over ShiftRegister and add SPI interface here (if that makes sense - haven't done the proper review yet, just quick skim)

Reply:
I didn't quite understand the second bullet could you please explain?